api/resid/resid.go

```diff
@@ -90,6 +90,13 @@ func (id ResId) GvknEquals(o ResId) bool {
 	return id.Name == o.Name && id.Gvk.Equals(o.Gvk)
 }
 
+// IsSelectedBy returns true if self is selected by the argument.
+func (id ResId) IsSelectedBy(selector ResId) bool {
+	return (selector.Name == "" || selector.Name == id.Name) &&
+		(selector.Namespace == "" || selector.IsNsEquals(id)) &&
```

Comment:
A question: what happens if someone uses `namespace: default`? Does the selector use the "effective namespace" and can there be a test for that?

Reply:
Took a quick look at the code actually, I see that it uses effective namespace

Reply:
yeah, that's why i changed it to use ResId.  So there's only one thing to fix if we get it wrong. 

There was lots of handwringing in the past around the handling of namespace since it can be totally undefined vs set as empty string vs set as the string "default" vs other things and these have different interpretations in different contexts.